src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java

```diff
@@ -256,11 +256,12 @@ private void doDeliverShadowTraffic(DeliveryRequest deliveryRequest) {
       // We need a clone here in order to safely modify the ClientInfo.
       DeliveryRequest requestToSend = deliveryRequest.clone();
       
+      Request request = requestToSend.getRequest();
       // We ensured earlier that client info was filled in.
-      assert requestToSend.getRequest().getClientInfo() != null;
+      assert request.getClientInfo() != null;
 
-      requestToSend.getRequest().getClientInfo().setClientType(ClientType.PLATFORM_SERVER);
-      requestToSend.getRequest().getClientInfo().setTrafficType(TrafficType.SHADOW);
+      request.setClientInfo(request.getClientInfo().toBuilder().setClientType(ClientType.PLATFORM_SERVER).build());
```

Comment:
I'm not sure why this is tobuilder+build+set twice in a row.

Is `request` a normal proto builder or still a plain old java object (POJO)?

If we switch Request fully to builders, this will look different.  We'd have a `Request.Builder` and then modify that.

If it's staying a POJO, then we can do

```
request.setClientInfo(request.getClientInfo().toBuilder()
    .setClientType(ClientType.PLATFORM_SERVER)
    .setTrafficType(TrafficType.SHADOW)
    .build())
```

Reply:
Combined the two lines